Runtime/Internal/DotNetWebSocket.cs

```diff
@@ -264,13 +264,21 @@ private async Task SendAsync()
         {
             while (_socket.State == System.Net.WebSockets.WebSocketState.Open)
             {
-                WebSocketMessage message;
+                WebSocketMessage message = null;
+                int count;
                 lock (_outgoingMessages)
                 {
-                    if (_outgoingMessages.Count == 0)
-                        continue;
+                    count = _outgoingMessages.Count;
+                    if (count != 0)
+                    {
+                        message = _outgoingMessages.Dequeue();
+                    }
+                }
 
-                    message = _outgoingMessages.Dequeue();
+                if (count == 0)
+                {
+                    await Task.Delay(10);
```

Comment:
Also, how did you arrive at 10 instead of e.g. 1 or 100? Arbitrary small number? Would be great to have a comment about how you arrived at the number.

Reply:
It was the only value I tested.
Thinking about it, I might try to remove the polling of messages to send and have the sending task be notified of new messages to send.